lib/split/persistence/cookie_adapter.rb

```diff
@@ -28,22 +29,25 @@ def keys
 
       private
 
-      def set_cookie(value)
-        @cookies[:split] = {
-          :value => JSON.generate(value),
-          :expires => @expires
-        }
+      def set_cookie(value = {})
+        @response.set_cookie :split.to_s, default_options.merge(value: JSON.generate(value))
```

Comment:
I think that this change may be causing split to send back multiple `Set-Cookie` headers in the case where a participant is involved in more than one split test. Version 3.0.0 did not have this problem. But after upgrading to 3.1.0 (which included this PR), I started seeing one `Set-Cookie`  response headers for every experiment mentioned in the cookie submitted by the client.

In a few cases, the cookies had enough experiments in them that my load balancer rejected the response as invalid.

Downgrading to 3.0.0 fixed the problem.

I think that `@response.set_cookie` is writing a new cookie each time it is called. However, `@cookies[:split]` was overwriting the previous value of the cookie in place.

Reply:
This demonstrates the issue: https://github.com/splitrb/split/pull/508